lib/src/io.dart

```diff
@@ -1132,3 +1132,21 @@ class PubProcessResult {
 
   bool get success => exitCode == exit_codes.SUCCESS;
 }
+
+/// Returns the top level directory in [path].
+///
+/// Throws an [ArgumentError] if [path] is just a filename with no directory.
+String topLevelDir(String pathString) {
```

Comment:
Instead of `path`, I'd do `uri` to make it clear this works on URI-formatted paths, not local file system ones.